### ISSUE TITLE
contains() does not keep scroll position if element with overflow:scroll is targeted

### ISSUE THREAD

**Issue by gabbersepp** [cypress-io/cypress#5788]
### Current behavior:

Given a html structure like this:

```html
<html>
    <style>
    #large {
        height: 500px;
        overflow:scroll
    }
    </style>
    <body>
        <div id="large">
            <p>1</p>
            <p>1</p>
.... many other p's
 <p id="toScroll">scroll</p>
</div></body></html>
```

and a spec like this:

```js
cy
            .get("p")
            .contains("scroll")
```

results in an unexpected behaviour.
After `contains` passes, the targeted `p` is scrolled out of the view.

As you can see, cypres scrolls to the element but then jumps to the scroll position it had just before the contains()

![gif](https://github.com/gabbersepp/cypress-example/blob/issue-5788/cypress-scroll.gif?raw=true)

### Desired behavior:

`contains` should ensure that the targeted element is scrolled into view after the command passes. 

### Steps to reproduce: (app code and test code)

repo: https://github.com/gabbersepp/cypress-example/tree/issue-5788

### Versions

Cypress 3.6.1, Win10

**Comment by gabbersepp** [comment]
OK I think I have misunderstood how scrolling works.
For those who run into the same trap:
- issue discussing "scrolling" during `get()`: https://github.com/cypress-io/cypress/issues/2353
- docu: https://docs.cypress.io/guides/core-concepts/interacting-with-elements.html#Scrolling

tl;dr:
Cypress does not scroll during simple get/find (this also holds for `contains`). During snapshot analysis, Cypress scrolls the elements into view to make it easier to debug a test.
